### PR TITLE
Allow saving state for multiple views in a single State (for MultiKeys)

### LINKED ISSUE #229: State for a MultiKey only stores the last View that save() was called for
`KeyDispatcher` passes `KeyChanger` a `Map` of contexts, but only one `incomingState`.  State.save() looks like this:
```
  public void save(@NonNull View view) {
    SparseArray<Parcelable> state = new SparseArray<>();
    view.saveHierarchyState(state);
    viewState = state;
  }
```

Unfortunately this means if you're saving view state of multiple keys, only the last one actually gets stored in the bundle 😭   Should probably have a `Map<Int, SparseArray<Parcelable>> viewState` instead and use view ids as the key?

### PR DESCRIPTION
**Pull request by edenman**
Fixes #229

### CODE REVIEW

File: flow/src/main/java/flow/State.java [hunk 1]
```diff
@@ -22,24 +22,41 @@
...
-    Object key = parceler.toKey(savedState.getParcelable("KEY"));
+    Object key = parceler.toKey(savedState.getParcelable(KEY));
     State state = new State(key);
-    state.viewState = savedState.getSparseParcelableArray("VIEW_STATE");
-    state.bundle = savedState.getBundle("BUNDLE");
+    int[] viewIds = checkNotNull(savedState.getIntArray(VIEW_STATE_IDS), "Null view state ids?");
+    for (int viewId : viewIds) {
+      SparseArray<Parcelable> viewState =
+          savedState.getSparseParcelableArray(VIEW_STATE_PREFIX + viewId);
+      if (viewState != null) {
```
Comment (loganj) [inline flow/src/main/java/flow/State.java]: Can this be null? If there's an id, seems like there should be a SparseArray for it.

File: flow/src/main/java/flow/State.java [hunk 2]
```diff
@@ -22,24 +22,41 @@
...
+      }
+    }
+    state.bundle = savedState.getBundle(BUNDLE);
     return state;
   }
 
   private final Object key;
   @Nullable private Bundle bundle;
-  @Nullable SparseArray<Parcelable> viewState;
+  // TODO shouldn't this be private?
```
Comment (loganj) [inline flow/src/main/java/flow/State.java]: Yeah, it should. Mind making the change?

File: flow/src/main/java/flow/State.java (hunk 1 above)
Comment (edenman) [inline flow/src/main/java/flow/State.java]: Might not be: the view could have `saveEnabled="false"` and there's this check in the `toBundle` method:
```
      SparseArray<Parcelable> viewState = entry.getValue();
      if (viewState.size() > 0) {
        outState.putSparseParcelableArray(VIEW_STATE_PREFIX + viewId, viewState);
      }
```

File: flow/src/main/java/flow/State.java (hunk 2 above)
Comment (edenman) [inline flow/src/main/java/flow/State.java]: done

File: flow/src/main/java/flow/State.java (hunk 1 above)
Comment (loganj) [inline flow/src/main/java/flow/State.java]: got it, thanks

**Pull request merged**